src/middlewared/middlewared/test/integration/assets/pool.py

```diff
@@ -28,19 +30,29 @@
 ]
 
 
+def generate_random_zpool_name():
+    zpool_name_length = 12
+    letters_and_digits = string.ascii_letters + string.digits
+    rando = ''.join(random.choices(letters_and_digits, k=zpool_name_length))
+    return f'cool_zpool_name_{rando}'
+
+
 @contextlib.contextmanager
-def another_pool(data=None, topology=None):
+def another_pool(name=None, data=None, topology=None):
```

Comment:
It'll have more sense to omit this parameter and just to `data.setdefault("name", generate_random_zpool_name())` as `data` is indeed the dictionary used for `pool.create`. But I also don't think random name will be useful at all, please see my comment above.